acvm-repo/acvm/src/compiler/transformers/mod.rs

```diff
@@ -79,8 +128,6 @@ pub(super) fn transform_internal<F: AcirField>(
                     &mut next_witness_index,
                 );
 
-                // Update next_witness counter
-                next_witness_index += (intermediate_variables.len() - len) as u32;
```

Comment:
Could you elaborate why this was removed again?

Reply:
Yes, I added it in the PR description:

> It is increased by the number of intermediate variables created [here](https://github.com/noir-lang/noir/blob/dc96de7c7a3d3c325b955edb277534653204cc5a/acvm-repo/acvm/src/compiler/transformers/mod.rs#L83) even though the mutable reference is [also increased](https://github.com/noir-lang/noir/blob/dc96de7c7a3d3c325b955edb277534653204cc5a/acvm-repo/acvm/src/compiler/transformers/csat.rs#L279-L281), hence the double increment

I might be missing the significance of this call here, but it looks like it's always incremented in tandem with any the actual variable being added. I guess this would have been needed if instead of passing a mutable reference, the method that adds the intermediate variables would increment their copy.